From 3f90eb34b13137666c6d21ed9dea11f9e85539aa Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 21 Sep 2021 12:30:28 +0800
Subject: [PATCH 26/30] Revert "drm/i915: Only access SFC_DONE when media
 domain is not fused off"

This reverts commit 65395b053d03cb662e63cbf2c7e0faef8c15cb8c.
---
 drivers/gpu/drm/i915/i915_gpu_error.c | 19 +------------------
 1 file changed, 1 insertion(+), 18 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gpu_error.c b/drivers/gpu/drm/i915/i915_gpu_error.c
index 9ce8f043ad7f..cf6e47adfde6 100644
--- a/drivers/gpu/drm/i915/i915_gpu_error.c
+++ b/drivers/gpu/drm/i915/i915_gpu_error.c
@@ -727,18 +727,9 @@ static void err_print_gt(struct drm_i915_error_state_buf *m,
 	if (INTEL_GEN(m->i915) >= 12) {
 		int i;
 
-		for (i = 0; i < GEN12_SFC_DONE_MAX; i++) {
-			/*
-			 * SFC_DONE resides in the VD forcewake domain, so it
-			 * only exists if the corresponding VCS engine is
-			 * present.
-			 */
-			if (!HAS_ENGINE(gt->_gt, _VCS(i * 2)))
-				continue;
-
+		for (i = 0; i < GEN12_SFC_DONE_MAX; i++)
 			err_printf(m, "  SFC_DONE[%d]: 0x%08x\n", i,
 				   gt->sfc_done[i]);
-		}
 
 		err_printf(m, "  GAM_DONE: 0x%08x\n", gt->gam_done);
 	}
@@ -1603,14 +1594,6 @@ static void gt_record_regs(struct intel_gt_coredump *gt)
 
 	if (INTEL_GEN(i915) >= 12) {
 		for (i = 0; i < GEN12_SFC_DONE_MAX; i++) {
-			/*
-			 * SFC_DONE resides in the VD forcewake domain, so it
-			 * only exists if the corresponding VCS engine is
-			 * present.
-			 */
-			if (!HAS_ENGINE(gt->_gt, _VCS(i * 2)))
-				continue;
-
 			gt->sfc_done[i] =
 				intel_uncore_read(uncore, GEN12_SFC_DONE(i));
 		}
-- 
2.27.0

